src/services/localSSHService.ts

```diff
@@ -12,21 +12,54 @@ import { IHostService } from './hostService';
 import SSHConfiguration from '../ssh/sshConfig';
 import { isWindows } from '../common/platform';
 import { getLocalSSHDomain } from '../remote';
-import { ITelemetryService } from './telemetryService';
+import { ITelemetryService, UserFlowTelemetry } from './telemetryService';
+import { ISessionService } from './sessionService';
 
 export interface ILocalSSHService {
+    flow?: UserFlowTelemetry;
     isSupportLocalSSH: boolean;
     initialized: Promise<void>;
 }
 
+type FailedToInitializeCode = 'RaceCondition' | 'UnavailableToImprove' | 'UserConfiguredWrong' | 'Unknown';
+
+// TODO: i18n support? 
+const FailedMsgRegexArr: Array<{ regexList: RegExp[]; failureCode: FailedToInitializeCode }> = [
```

Comment:
Could we rather have original error codes, like `noSpaceLeft`, `eperm`, `eaccess`, `sshConfigDontExist` and so on. Analytics don't have high cardinality issues and it would be good to preserve original errors.

Reply:
Both `fs error` and `vscode.FileSystemError` contain `code` property, going to use it